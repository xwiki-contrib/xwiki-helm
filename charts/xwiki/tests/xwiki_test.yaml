suite: test deployment
templates:
  - xwiki.yaml
  - configmaps.yaml
  - initialization-configmaps.yaml
tests:
  - it: testing default values
    documentIndex: 0
    asserts:
      - isKind:
          of: StatefulSet
        template: xwiki.yaml
      - isNotEmpty:
          path: spec.template.metadata.annotations["checksum/configmap"]
        template: xwiki.yaml
      - isNotEmpty:
          path: spec.template.metadata.annotations["checksum/initialization-configmaps"]
        template: xwiki.yaml
  - it: should use custom image
    set:
      image.tag: latest
      mysql.enabled: false
      postgresql.enabled: false
      mariadb.enabled: false
      image.name: default
      workloadStateful: false
    asserts:
      - isKind:
          of: Deployment
        template: xwiki.yaml
        documentIndex: 0
      - equal:
          path: spec.template.spec.containers[0].image
          value: default:latest
        template: xwiki.yaml
        documentIndex: 0
      - matchSnapshot:
          path: spec.template.spec.containers[0].env
        template: xwiki.yaml
        documentIndex: 0
  - it: should use mysql lts image
    set:
      mysql.enabled: true
      postgresql.enabled: false
      mariadb.enabled: false
    asserts:
      - equal:
          path: spec.template.spec.containers[0].image
          value: xwiki:lts-mysql-tomcat
        template: xwiki.yaml
      - matchSnapshot:
          path: spec.template.spec.containers[0].env
        template: xwiki.yaml
  - it: should use postgres lts image
    set:
      mysql.enabled: false
      postgresql.enabled: true
      mariadb.enabled: false
    asserts:
      - equal:
          path: spec.template.spec.containers[0].image
          value: xwiki:lts-postgres-tomcat
        template: xwiki.yaml
      - matchSnapshot:
          path: spec.template.spec.containers[0].env
        template: xwiki.yaml
  - it: should properly set value
    set:
      image.pullPolicy: IfNotPresent
      service.internalPort: 8080
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: IfNotPresent
        template: xwiki.yaml
      - equal:
          path: spec.template.spec.containers[0].ports[0].containerPort
          value: 8080
        template: xwiki.yaml
#      - equal:
#          path: spec.template.spec.containers[0].volumeMounts[1].name
#          value: xwiki-data
#        template: xwiki.yaml
#      - equal:
#          path: spec.template.spec.volumes[1].name
#          value: xwiki-data
#        template: xwiki.yaml
#      - equal:
#          path: spec.template.spec.volumes[1].persistentVolumeClaim.claimName
#          value: xwiki
#        template: xwiki.yaml
#      - matchSnapshot:
#          path: spec.template.spec.containers[1].volumeMounts
#        template: xwiki.yaml
#      - matchSnapshot:
#          path: spec.template.spec.volumes
#        template: xwiki.yaml
#  - it: should properly set value pvc
#    asserts:
#      - isKind:
#          of: PersistentVolumeClaim
#        documentIndex: 1
#      - matchSnapshot:
#          path: spec
#        documentIndex: 1
#      - equal:
#          path: metadata.name
#          value: xwiki
#        documentIndex: 1
  - it: should use custom DB password secret
    set:
      externalDB.customKeyRef.enabled: true
      externalDB.customKeyRef.name: "custom-secret"
      externalDB.customKeyRef.key: "custom-key"
    asserts:
      - matchSnapshot:
          path: spec.template.spec.containers[0].env
        template: xwiki.yaml
  - it: should create a postgres init container
    set:
      mysql.enabled: false
      postgresql.enabled: true
      mariadb.enabled: false
      initContainers.database.enabled: true
      postgresql.image.repository: "postgres"
      postgresql.image.tag: "latest"
    asserts:
      - matchSnapshot:
          path: spec.template.spec.initContainers
        template: xwiki.yaml
  - it: should create a mariadb init container
    set:
      mysql.enabled: false
      postgresql.enabled: false
      mariadb.enabled: true
      initContainers.database.enabled: true
      mariadb.image.repository: "mariadb"
      mariadb.image.tag: "latest"
    asserts:
      - matchSnapshot:
          path: spec.template.spec.initContainers
        template: xwiki.yaml
  - it: should create a mysql init container
    set:
      mysql.enabled: true
      postgresql.enabled: false
      mariadb.enabled: false
      initContainers.database.enabled: true
      mysql.image.repository: "mysql"
      mysql.image.tag: "latest"
    asserts:
      - matchSnapshot:
          path: spec.template.spec.initContainers
        template: xwiki.yaml
  - it: should create a solr init container
    set:
      solr.enabled: true
      initContainers.solr.enabled: true
    asserts:
      - matchSnapshot:
          path: spec.template.spec.initContainers
        template: xwiki.yaml
  - it: should create an external db init container for mysql
    set:
      externalDB.customKeyRef.enabled: true
      externalDB.customKeyRef.name: "custom-secret"
      externalDB.customKeyRef.key: "custom-key"
      externalDB.type: "mysql"
      initContainers.database.enabled: true
      initContainers.database.image: "mysql:latest"
    asserts:
      - matchSnapshot:
          path: spec.template.spec.initContainers
        template: xwiki.yaml
