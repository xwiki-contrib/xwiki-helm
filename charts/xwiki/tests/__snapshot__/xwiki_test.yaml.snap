should create a mariadb init container:
  1: |
    - args:
        - |
          for i in $(seq 1 30); do
            if eval $CHECK_DB; then
              echo "Database is ready!"
              exit 0
            fi
            echo "Waiting for database..."
            sleep 1
          done
          echo "Database is not ready!"
          exit 1
      command:
        - /bin/sh
        - -ec
      env:
        - name: DB_PASSWORD
          valueFrom:
            secretKeyRef:
              key: DB_PASSWORD
              name: RELEASE-NAME
        - name: DB_HOST
          valueFrom:
            configMapKeyRef:
              key: DB_HOST
              name: RELEASE-NAME-xwiki
        - name: DB_USER
          valueFrom:
            configMapKeyRef:
              key: DB_USER
              name: RELEASE-NAME-xwiki
        - name: DB_DATABASE
          valueFrom:
            configMapKeyRef:
              key: DB_DATABASE
              name: RELEASE-NAME-xwiki
        - name: CHECK_DB
          value: mysqladmin ping -h $DB_HOST -u $DB_USER -p$DB_PASSWORD
      image: mariadb:latest
      name: wait-for-db
should create a mysql init container:
  1: |
    - args:
        - |
          for i in $(seq 1 30); do
            if eval $CHECK_DB; then
              echo "Database is ready!"
              exit 0
            fi
            echo "Waiting for database..."
            sleep 1
          done
          echo "Database is not ready!"
          exit 1
      command:
        - /bin/sh
        - -ec
      env:
        - name: DB_PASSWORD
          valueFrom:
            secretKeyRef:
              key: mysql-password
              name: RELEASE-NAME-mysql
        - name: DB_HOST
          valueFrom:
            configMapKeyRef:
              key: DB_HOST
              name: RELEASE-NAME-xwiki
        - name: DB_USER
          valueFrom:
            configMapKeyRef:
              key: DB_USER
              name: RELEASE-NAME-xwiki
        - name: DB_DATABASE
          valueFrom:
            configMapKeyRef:
              key: DB_DATABASE
              name: RELEASE-NAME-xwiki
        - name: CHECK_DB
          value: mysqladmin ping -h $DB_HOST -u $DB_USER -p$DB_PASSWORD
      image: mysql:latest
      name: wait-for-db
should create a postgres init container:
  1: |
    - args:
        - |
          for i in $(seq 1 30); do
            if eval $CHECK_DB; then
              echo "Database is ready!"
              exit 0
            fi
            echo "Waiting for database..."
            sleep 1
          done
          echo "Database is not ready!"
          exit 1
      command:
        - /bin/sh
        - -ec
      env:
        - name: DB_PASSWORD
          valueFrom:
            secretKeyRef:
              key: password
              name: RELEASE-NAME-postgresql
        - name: DB_HOST
          valueFrom:
            configMapKeyRef:
              key: DB_HOST
              name: RELEASE-NAME-xwiki
        - name: DB_USER
          valueFrom:
            configMapKeyRef:
              key: DB_USER
              name: RELEASE-NAME-xwiki
        - name: DB_DATABASE
          valueFrom:
            configMapKeyRef:
              key: DB_DATABASE
              name: RELEASE-NAME-xwiki
        - name: CHECK_DB
          value: PGPASSWORD=$DB_PASSWORD pg_isready -h $DB_HOST -U $DB_USER -d $DB_DATABASE
      image: postgres:latest
      name: wait-for-db
should create a solr init container:
  1: |
    - command:
        - /bin/sh
        - -ec
        - |
          for i in $(seq 1 30); do
            if curl --silent --connect-timeout "15000" $SOLR_BASEURL/admin/info/system | grep '\"status\":0'
              echo "Solr is ready!"
              exit 0
            fi
            echo "Waiting for Solr..."
            sleep 1
          done
          echo "Solr is not ready!"
          exit 1
      env:
        - name: SOLR_BASEURL
          valueFrom:
            configMapKeyRef:
              key: SOLR_BASEURL
              name: RELEASE-NAME-xwiki
      image: alpine/curl:8.9.0
      name: wait-for-solr
should create an external db init container for mysql:
  1: |
    - args:
        - |
          for i in $(seq 1 30); do
            if eval $CHECK_DB; then
              echo "Database is ready!"
              exit 0
            fi
            echo "Waiting for database..."
            sleep 1
          done
          echo "Database is not ready!"
          exit 1
      command:
        - /bin/sh
        - -ec
      env:
        - name: DB_PASSWORD
          valueFrom:
            secretKeyRef:
              key: custom-key
              name: custom-secret
        - name: DB_HOST
          valueFrom:
            configMapKeyRef:
              key: DB_HOST
              name: RELEASE-NAME-xwiki
        - name: DB_USER
          valueFrom:
            configMapKeyRef:
              key: DB_USER
              name: RELEASE-NAME-xwiki
        - name: DB_DATABASE
          valueFrom:
            configMapKeyRef:
              key: DB_DATABASE
              name: RELEASE-NAME-xwiki
        - name: CHECK_DB
          value: mysqladmin ping -h $DB_HOST -u $DB_USER -p$DB_PASSWORD
      image: mysql:latest
      name: wait-for-db
should use custom DB password secret:
  1: |
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: custom-key
          name: custom-secret
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
should use custom image:
  1: |
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: DB_PASSWORD
          name: RELEASE-NAME
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
should use mysql lts image:
  1: |
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: mysql-password
          name: RELEASE-NAME-mysql
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
should use postgres lts image:
  1: |
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: password
          name: RELEASE-NAME-postgresql
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
